### DISCUSSION TITLE
Setting the state of a Select component that has it's options dynamically set doesn't work

### DISCUSSION THREAD

**Discussion by zealouscreations** [filamentphp/filament#14483] [Help]
### Package

Form builder

### Package Version

v3.2.113

### How can we help you?

This is in the context of making a form in an `Action`

I have two `Select` components. The **first `Select`** will have options of models using the `id` for the value and the `name` for the label.

When a selection is made on the **first `Select`** the `afterStatUpdated()` will populate the `options` of the **second `Select`** and sets the value of the **second `Select`** component to the selected model's `default_option`, but it doesn't update the select dropdown to that value.

Here is a code example of what I've tried so far:

```php
Action::make('Dynamic Selection Example')
    ->form([
        Select::make('some_model_id')
            ->options(SomeModel::get()->pluck('name', 'id')->toArray())
            ->afterStateUpdated(function($state, Select $component, Set $set) {
                $someModel = SomeModel::find($state);
                $component->getContainer()->getComponent('some_option')->options(collect($someModel?->options)->pluck('name', 'name')->toArray()); // <-- Tried this way second, and still doesn't work
                $set('some_option', $someModel?->default_option);
                $component->getContainer()->getComponent('test')->state(fn(Get $get) => 'selected option: '.$get('some_option'));
            })
            ->required()
            ->live(),
        Select::make('some_option')->key('some_option')
//            ->options(fn(Get $get) => collect(SomeModel::find($get('some_model_id'))?->options)->pluck('name', 'name')->toArray()) // <-- Tried this way first
            ->required(),
        TextInput::make('test')->label('Test')->key('test')->disabled(), // <-- This will display the expected value set from the afterStateUpdated callback
    ]);
```

I've tried many, many ways to get this to work, to no avail. I realize that the timing of the `options` of the **second `Select`** getting rendered might not be ready when the value of the `Select` is set, but there is no `afterOptionsRendered()` (or something like that) to hook into to set the value after the dynamically generated options are rendered.

**Comment by Lirux** [comment]
Hi @zealouscreations 

Check this out: https://github.com/filamentphp/filament/discussions/12074

This worked for me.
